cmd/sync-tags/gomod.go

```diff
@@ -31,7 +32,7 @@ import (
 )
 
 // updateGomodWithTaggedDependencies gets the dependencies at the given tag and fills go.mod and go.sum.
-// If anything is changed, it commits the changes. Returns true if go.mod changed.
+// Returns true if go.mod has been successfully changed in root and any submodules if it exists.
 func updateGomodWithTaggedDependencies(tag string, depsRepo []string, semverTag bool) (bool, error) {
```

Comment:
is this run once per staging repo, in topological order? specifically, is this run on the code-organization repo before the other staging repos are processed?

Reply:
> is this run once per staging repo, in topological order? 

yes:

https://github.com/kubernetes/publishing-bot/blob/1b0ee0a68aa88355f6248044be28be8166bb15ad/artifacts/scripts/construct.sh#L147

@akhilerm in addition to sync-tags, we'll also need to fix normal publishing (not just tags) in https://github.com/kubernetes/publishing-bot/blob/master/artifacts/scripts/util.sh

Reply:
> > is this run once per staging repo, in topological order?
> 
> yes:

Then processing the submodules at the same time as the parent module won't be correct, since code-generation runs first, but its submodule references other modules like client-go / api / apimachinery, etc.

I'm really on the fence whether it is a good idea to try to resolve/pin the require directives in these test/example submodules, but if we do, we shouldn't just process them as another normal staging repo and make them usable as libraries externally